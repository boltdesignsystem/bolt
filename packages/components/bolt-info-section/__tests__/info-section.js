// Refer to https://github.com/boltdesignsystem/bolt/wiki/Jest-Test:-Example-Jest-Test for more testing examples
import { render, stopServer } from '../../../testing/testing-helpers';
// import schema from '../info-section.schema';
// const { [PROP KEY], [PROP KEY]... } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
});

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    label: {
      content: 'This is an info section',
      tag: 'h2',
    },
    content:
      'This is the section content. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Fugit labore hic officia maiores ut sequi ipsam nihil repudiandae numquam quidem, molestiae amet aliquid sapiente sed odio quo aspernatur mollitia a.',
    details_link: {
      label: 'This is the details link',
      attributes: {
        type: 'button',
      },
    },
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt Info Section component', () => {
  test('default', async () => {
    const results = await render(
      '@bolt-components-info-section/info-section.twig',
      {
        ...fixtures.defaultData,
      },
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

// describe('Bolt Info Section prop -', () => {
//   // Target each of the schema keys with the following pattern
//   [PROP KEY].enum.forEach(async (option) => {
//     test(`[PROP KEY] items: ${option}`, async () => {
//       const results = await render(
//         '@bolt-components-info-section/info-section.twig',
//         {
//           ...fixtures.defaultData,
//           [PROP KEY]: option,
//         }
//       );

//       await expect(results.ok).toBe(true);
//       await expect(results.html).toMatchSnapshot();
//     });
//   });
// });
