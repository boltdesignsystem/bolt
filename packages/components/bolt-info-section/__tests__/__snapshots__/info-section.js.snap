// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Info Section component default 1`] = `
<div class="c-bolt-info-section">
  <h2 class="c-bolt-info-section__headline">
    This is an info section
  </h2>
  <button class="c-bolt-info-section__headline c-bolt-info-section__headline--button js-bolt-info-section-toggle-button"
          type="button"
          aria-expanded="false"
  >
    This is an info section
    <span class="c-bolt-info-section__headline-icon">
      <svg class="e-bolt-icon"
           aria-hidden="true"
           enable-background="new 0 0 32 32"
           viewbox="0 0 32 32"
           xmlns="http://www.w3.org/2000/svg"
      >
        <path clip-rule="evenodd"
              d="m26.6 12.4c.6-.6.6-1.6 0-2.2s-1.6-.6-2.2 0l-8.4 8.3-8.3-8.4c-.6-.6-1.6-.6-2.2 0-.3.4-.5.8-.5 1.2s.2.8.5 1.1l9.4 9.4c.6.6 1.6.6 2.2 0z"
              fill="#151619"
              fill-rule="evenodd"
        >
        </path>
      </svg>
    </span>
  </button>
  <div class="c-bolt-info-section__content">
    This is the section content. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Fugit labore hic officia maiores ut sequi ipsam nihil repudiandae numquam quidem, molestiae amet aliquid sapiente sed odio quo aspernatur mollitia a.
  </div>
  <div class="c-bolt-info-section__link">
    <button type="button"
            class="e-bolt-text-link e-bolt-text-link--reversed-underline e-bolt-text-link--expand-click-target"
    >
      This is the details link
      <span class="e-bolt-text-link__icon-after"
            aria-hidden="true"
      >
        <svg class="e-bolt-icon"
             aria-hidden="true"
             enable-background="new 0 0 32 32"
             viewbox="0 0 32 32"
             xmlns="http://www.w3.org/2000/svg"
        >
          <path clip-rule="evenodd"
                d="m12.4 5.4c-.6-.6-1.6-.6-2.2 0s-.6 1.6 0 2.3l8.4 8.3-8.3 8.3c-.6.6-.6 1.6 0 2.2.3.3.7.5 1 .5.4 0 .8-.2 1.1-.5l9.4-9.4c.6-.6.6-1.6 0-2.2z"
                fill="#151619"
                fill-rule="evenodd"
          >
          </path>
        </svg>
      </span>
    </button>
  </div>
</div>
`;
